src/payload.rs

```diff
@@ -481,7 +480,13 @@ impl Iterator for ZBytesInIterator {
         let mut data = MaybeUninit::<z_owned_bytes_t>::uninit();
```

Comment:
we likely need here to add return value to iter_body to tell when we are finished if we want to consider empty bytes as valid state